samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -568,6 +540,10 @@ object SamzaContainer extends Logging {
         new StorageConfig(config).getChangeLogDeleteRetentionsInMs,
         new SystemClock)
 
+      val tableManager = new TableManager(config, taskStores.asJava)
```

Comment:
How about potential remote stores? file-based table providers? I think that introducing an init() method to initialize all tableProviders after constructor of tableManager would seem more general approach than having various different types of constructors of TableManager.

Reply:
Fixed, see above.